mazegame/Assets/Scripts/Maze/MazeGenerator.cs

```diff
@@ -0,0 +1,176 @@
+        }
+
+        maze = new Maze(mazeWidth, mazeHeight, mazeRG);
+        maze.Generate();
+
+        mazeGoalPosition = maze.GetGoalPosition();
+
+        DrawMaze();
+
+        if (OnMazeReady != null)
+        {
+            OnMazeReady();
+        }
+    }
+
+    void DrawMaze()
+    {
+        for (int x = 0; x < mazeWidth; x++)
+        {
+            for (int y = 0; y < mazeHeight; y++)
+            {
+                Vector3 position = new Vector3(x, y);
+
+                if (maze.Grid[x, y] == true)
+                {
+                    CreateMazeSprite(position, floorSprite, transform, 0, mazeRG.Next(0, 3) * 90);
+                }
+                else
+                {
+                    CreateMazeSprite(position, roofSprite, transform, 0, 0);
+
+                    DrawWalls(x, y);
+                }
+
+            }
+        }
+    }
+
+    void CreateMazeSprite(Vector3 position, Sprite sprite, Transform parent, int sortingOrder, float rotation)
+    {
+        MazeSprite mazeSprite = Instantiate(mazeSpritePrefab, position, Quaternion.identity) as MazeSprite;
+        mazeSprite.SetSprite(sprite, sortingOrder);
+        mazeSprite.transform.SetParent(parent);
+        mazeSprite.transform.Rotate(0, 0, rotation);
+    }
+
+    void DrawWalls(int x, int y)
+    {
+        bool top = GetMazeGridCell(x, y + 1);
+        bool bottom = GetMazeGridCell(x, y - 1);
+        bool right = GetMazeGridCell(x + 1, y);
+        bool left = GetMazeGridCell(x - 1, y);
+
+        Vector3 position = new Vector3(x, y);
+
+        if (top)
+        {
+            CreateMazeSprite(position, wallSprite, transform, 1, 0);
+        }
+
+        if (left)
+        {
+            CreateMazeSprite(position, wallSprite, transform, 1, 90);
+        }
+
+        if (bottom)
+        {
+            CreateMazeSprite(position, wallSprite, transform, 1, 180);
+        }
+
+        if (right)
+        {
+            CreateMazeSprite(position, wallSprite, transform, 1, 270);
+        }
+
+        if (!left && !top && x > 0 && y < mazeHeight - 1)
+        {
+            CreateMazeSprite(position, wallCornerSprite, transform, 2, 0);
+        }
+
+        if (!left && !bottom && x > 0 && y > 0)
+        {
+            CreateMazeSprite(position, wallCornerSprite, transform, 2, 90);
+        }
+
+        if (!right && !bottom && x < mazeWidth - 1 && y > 0)
+        {
+            CreateMazeSprite(position, wallCornerSprite, transform, 2, 180);
+        }
+
+        if (!right && !top && x < mazeWidth - 1 && y < mazeHeight - 1)
+        {
+            CreateMazeSprite(position, wallCornerSprite, transform, 2, 270);
+        }
+    }
+
+    public bool GetMazeGridCell(int x, int y)
+    {
+        return maze.GetCell(x, y);
+    }
+
+    public static MazeGenerator instance;
+
+    void Awake()
+    {
+        instance = this;
+    }
+
+    public List<Vector3> GetRandomFloorPositions(int count)
+    {
+        List<Vector3> positions = new List<Vector3>();
```

Comment:
`floorPositions` would be a more appropriate name

Reply:
changed